src/layout/mod.rs

```diff
@@ -719,6 +784,34 @@ impl<W: LayoutElement> Layout<W> {
         None
     }
 
+    pub fn find_workspace_by_name(&self, workspace_name: &str) -> Option<(usize, &Workspace<W>)> {
+        if let MonitorSet::Normal { ref monitors, .. } = &self.monitor_set {
+            for mon in monitors {
+                if let Some((index, workspace)) = mon
+                    .workspaces
+                    .iter()
+                    .enumerate()
+                    .find(|(_, w)| w.name == Some(workspace_name.to_string()))
+                {
+                    return Some((index, workspace));
+                }
+            }
+        }
+
+        None
+    }
+
+    pub fn unname_workspace(&mut self, workspace_name: &str) {
+        if let MonitorSet::Normal {
+            ref mut monitors, ..
+        } = &mut self.monitor_set
+        {
+            for mon in monitors {
+                mon.unname_workspace(workspace_name);
```

Comment:
After this it should run clean up workspaces, as unnamed empty workspaces need to be removed.

You should add an unname operation to proptest below, it would've caught this for you.

Reply:
I wanted to add a proptest, but after about an hour of trying to wrap my head around them, I gave up. :(